pkg/app/ops/firestoreindexcreator/indexcreator.go

```diff
@@ -0,0 +1,64 @@
+// Copyright 2021 The PipeCD Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package firestoreindexcreator automatically creates the needed composite indexes for Google Firestore,
+// based on well-defined JSON format indexes list with the name indexes.json.
+package firestoreindexcreator
+
+import (
+	"context"
+	"fmt"
+
+	"go.uber.org/zap"
+)
+
+type IndexCreator interface {
```

Comment:
We may need to delete some unused indexes so maybe `Ensurer` is better than `Creator`.

Reply:
The package should be `firestoreindexensurer` as well right.

Reply:
Right. 👍